slasher/db/iface/interface.go

```diff
@@ -24,8 +24,8 @@ type ReadOnlyDatabase interface {
 	GetLatestEpochDetected(ctx context.Context) (types.Epoch, error)
 
 	// BlockHeader related methods.
-	BlockHeaders(ctx context.Context, slot types.Slot, validatorID uint64) ([]*ethpb.SignedBeaconBlockHeader, error)
-	HasBlockHeader(ctx context.Context, slot types.Slot, validatorID uint64) bool
+	BlockHeaders(ctx context.Context, slot types.Slot, validatorID types.ValidatorIndex) ([]*ethpb.SignedBeaconBlockHeader, error)
```

Comment:
Can we change `ID` to `index` in this file's interfaces? They are not really the same and we use indexes here. This might be too big of a change for this PR, though.

Reply:
Also I wonder whether it would not be a braking change...

Reply:
it should indeed be validatorIdx or validatorIndex

Reply:
Sounds good. Will do that in the next PR. It seems out of scope for now